src/main/java/com/squareup/javapoet/ClassName.java

```diff
@@ -122,7 +122,7 @@ public static ClassName bestGuess(String classNameString) {
 
     // Add the package name, like "java.util.concurrent", or "" for no package.
     int p = 0;
-    while (p < classNameString.length() && Util.isLowerCase(classNameString.charAt(p))) {
+    while (p < classNameString.length() && Character.isLowerCase(classNameString.charAt(p))) {
```

Comment:
In practice you might face with runtime class loading errors if you use
such letters in class names.

On Tue Jan 27 2015 at 6:23:02 PM Éamonn McManus <notifications@github.com>
wrote:

> In src/main/java/com/squareup/javapoet/ClassName.java
> <https://github.com/square/javapoet/pull/208#discussion_r23620051>:
>
> > @@ -122,7 +122,7 @@ public static ClassName bestGuess(String classNameString) {
> >
> >      // Add the package name, like "java.util.concurrent", or "" for no package.
> >      int p = 0;
> > -    while (p < classNameString.length() && Util.isLowerCase(classNameString.charAt(p))) {
> > +    while (p < classNameString.length() && Character.isLowerCase(classNameString.charAt(p))) {
>
> As an experiment, in NetBeans I wrote this in a method:
> int \ud835\udc00 = 5;
> int 𝐀 = 5;
> It correctly complained that the second 𝐀 was a redefinition of the first
> (hidden in a surrogate pair). :-)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/square/javapoet/pull/208/files#r23620051>.
>